src/multiselect.js

```diff
@@ -0,0 +1,206 @@
+/**
+ * @license
+ * Copyright 2022 MIT
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Plugin main class.
+ */
+
+import * as Blockly from 'blockly/core';
+
+import * as ContextMenu from './multiselect_contextmenu';
+import * as Shortcut from './multiselect_shortcut';
+import {blockSelection, inMultipleSelectionMode,
+  hasSelectedParent, setBaseBlockDragger} from './global';
+import {MultiselectControls} from './multiselect_controls';
+
+/**
+ * Class for using multiple select blocks on workspace.
+ */
+export class Multiselect {
+  /**
+   * Initalize the class data structure.
+   * @param {!Blockly.WorkspaceSvg} workspace The workspace to sit in.
+   */
+  constructor(workspace) {
+    this.workspace_ = workspace;
+    this.origHandleWsStart_ = Blockly.Gesture.prototype.handleWsStart;
+  }
+
+  /**
+   * Bind the events and replace registration.
+   * @param {{multiselectIcon: Object, useDoubleClick: boolean}} options
+   * to set.
+   */
+  init(options) {
+    this.onKeyDownWrapper_ = Blockly.browserEvents.conditionalBind(
+        this.workspace_.getInjectionDiv(), 'keydown', this, this.onKeyDown_);
+    this.onKeyUpWrapper_ = Blockly.browserEvents.conditionalBind(
+        this.workspace_.getInjectionDiv(), 'keyup', this, this.onKeyUp_);
+    this.eventListenerWrapper_ = this.eventListener_.bind(this);
+    this.workspace_.addChangeListener(this.eventListenerWrapper_);
+
+    ContextMenu.unregisterContextMenu();
+    ContextMenu.registerOurContextMenu();
+    Shortcut.unregisterShortcut();
+    Shortcut.registerOurShortcut();
```

Comment:
What happens if we initialize the plugin multiple times? In App Inventor we have one workspace per screen, so I expect that means we will have one copy of the plugin per workspace. Verify that the code here works correctly if called multiple times.

Reply:
Will check that with https://github.com/mit-cml/workspace-multiselect/pull/1#discussion_r929661470 together.

Reply:
Now make it to check if `Blockly.ContextMenuRegistry.registry.registry_.workspaceSelectAll` exists, then will not register twice. This would avoid the issue and I've verified by initializing the plugin multiple times